src/frontend/Ast.ml

```diff
@@ -8,6 +8,9 @@ type identifier =
   {name: string; id_loc: Location_span.t sexp_opaque [@compare.ignore]}
 [@@deriving sexp, hash, compare]
 
+type comment = LineComment of string | MultiComment of string list
```

Comment:
Did you consider making comments just list of strings and deciding the format upon output? That would also save us from needing lists of comments later on, but I understand it would lose some of the original style.